src/analytics/helpers.js

```diff
@@ -0,0 +1,94 @@
+import moment from 'moment'
+import { links } from 'config'
+import formatFacetNames from 'helpers/formatFacetNames'
+import formatFilterName from 'helpers/formatFilterName'
+import getReadable from 'helpers/getReadable'
+
+const datasetOptionsKeys = ['aggregate_by', 'scale_by', 'quantile_normalize']
+
+// adds a custom event to GA4
+// https://developers.google.com/analytics/devguides/collection/ga4/events?client_type=gtag
+export const event = (eventName, value = {}, nonInteraction = false) => {
+  window.gtag('event', eventName, {
+    ...value,
+    non_interaction: nonInteraction
+  })
+}
+
+// formats a given string to snake_case for event parameters
+export const getSnakeCase = (str) =>
+  str.replace(/([a-z])([A-Z])/g, '$1_$2').toLowerCase()
```

Comment:
I've added this utility helper method to `analytics/helpers` to convert component names from camelCase to snake_case that is the accepted format for GA event parameters. Should we keep it here or move it to `helpers`?

Reply:
Since we eliminated the use of template strings for dynamically assigning parameter names and now define them explicitly, the `getSnakeCase` method is no longer necessary. Thus, it has been removed, see at d365c507d1043e21b96038c9d7b5c8a9c87d4882.